src/services/promiseUtilsService.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+angular.module('openshiftCommonServices')
+  .factory('PromiseUtils', function($q) {
+    return {
+      // Returns a promise that is resolved or rejected only after all promises
+      // complete. `promises` is a collection of promises. `null` or
+      // `undefined` values are treated as "complete."
+      //
+      // Different than `$q.all` in that it will always wait for all promises.
+      // `$q.all` short circuits as soon as one fails.
+      waitForAll: function(promises) {
+        var total = _.size(promises);
+        if (!total) {
+          return $q.when();
+        }
+
+        var deferred = $q.defer();
+        var complete = 0;
+        var failed = false;
+        var checkDone = function() {
+          if (complete < total) {
+            return;
+          }
+
+          if (failed) {
+            deferred.reject();
```

Comment:
i think $q.all passes up the content from the failure when it does a reject, not positive.  Not that we necessarily need that, but we should document above if it is a difference.

Reply:
Thanks, updated the comment.